npc/ipblock.go

```diff
@@ -0,0 +1,129 @@
+package npc
+
+import (
+	"fmt"
+	"github.com/weaveworks/weave/common"
+	"github.com/weaveworks/weave/npc/ipset"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net"
+	"sort"
+	"strings"
+)
+
+type ipBlock struct {
+	ipsetName ipset.Name // generated ipset name
+	spec      *networkingv1.IPBlock
+}
+
+func newIPBlock(ipb *networkingv1.IPBlock, ns string) (b *ipBlock, err error) {
+	_, _, err = net.ParseCIDR(ipb.CIDR)
+	if err != nil {
+		return
+	}
+
+	for _, ex := range ipb.Except {
+		_, _, err = net.ParseCIDR(ex)
+		if err != nil {
+			return
+		}
+	}
+
+	b = &ipBlock{spec: ipb}
+
+	if len(ipb.Except) > 0 {
+		sortedExcept := ipb.Except
+		sort.Strings(sortedExcept)
+		b.ipsetName = ipset.Name(IpsetNamePrefix + shortName("except:"+strings.Join(sortedExcept, "")))
```

Comment:
What is a value of `ipsetName` when `ipb.Except` is empty?

Reply:
It is empty because no ipset will be created if `ipb.Except` is empty. I've refactored some code and will update this PR along with more test.